GPU/TPCFastTransformation/Spline1DSpec.h

```diff
@@ -0,0 +1,528 @@
+
+  /// Get Xmin
+  GPUd() DataT getXmin() const { return mXmin; }
+
+  /// Get XtoUscale
+  GPUd() DataT getXtoUscale() const { return mXtoUscale; }
+
+  /// Set X range
+  void setXrange(DataT xMin, DataT xMax);
+
+  /// Print method
+  void print() const;
+
+  ///  _______________  Expert tools  _______________
+
+  /// Number of parameters for given Y dimensions
+  GPUd() int calcNumberOfParameters(int nYdim) const { return (2 * nYdim) * getNumberOfKnots(); }
+
+  ///_______________  Test tools  _______________
+
+#if !defined(GPUCA_GPUCODE) && !defined(GPUCA_STANDALONE) // code invisible on GPU and in the standalone compilation
+  /// Test the class functionality
+  static int test(const bool draw = 0, const bool drawDataPoints = 1);
+#endif
+
+  /// _____________  FlatObject functionality, see FlatObject class for description  ____________
+
+  using FlatObject::getBufferAlignmentBytes;
+  using FlatObject::getClassAlignmentBytes;
+
+#if !defined(GPUCA_GPUCODE)
+  void cloneFromObject(const Spline1DContainer& obj, char* newFlatBufferPtr);
+  void moveBufferTo(char* newBufferPtr);
+#endif
+
+  using FlatObject::releaseInternalBuffer;
+
+  void destroy();
+  void setActualBufferAddress(char* actualFlatBufferPtr);
+  void setFutureBufferAddress(char* futureFlatBufferPtr);
+
+ protected:
+  /// Non-const accessor to the knots array
+  Knot* getKnots() { return reinterpret_cast<Knot*>(mFlatBufferPtr); }
+
+  /// Non-const accessor to U->knots map
+  int* getUtoKnotMap() { return mUtoKnotMap; }
+
+#if !defined(GPUCA_GPUCODE)
+  /// Constructor for a regular spline
+  void recreate(int nYdim, int numberOfKnots);
+
+  /// Constructor for an irregular spline
+  void recreate(int nYdim, int numberOfKnots, const int knotU[]);
+#endif
+
+  /// _____________  Data members  ____________
+
+  int mYdim = 0;                ///< dimentionality of F
+  int mNumberOfKnots = 0;       ///< n knots on the grid
+  int mUmax = 0;                ///< U of the last knot
+  DataT mXmin = 0;              ///< X of the first knot
+  DataT mXtoUscale = 0;         ///< a scaling factor to convert X to U
+  int* mUtoKnotMap = nullptr;   //! (transient!!) pointer to (integer U -> knot index) map inside the mFlatBufferPtr array
+  DataT* mParameters = nullptr; //! (transient!!) pointer to F-dependent parameters inside the mFlatBufferPtr array
+
+#ifndef GPUCA_ALIROOT_LIB
+  ClassDefNV(Spline1DContainer, 1);
+#endif
+};
+
+template <typename DataT>
+template <typename Spline1DContainer<DataT>::SafetyLevel SafeT>
+GPUdi() int Spline1DContainer<DataT>::getLeftKnotIndexForU(DataT u) const
+{
+  /// Get i: u is in [knot_i, knot_{i+1}) segment
+  /// when u is otside of [0, mUmax], return a corresponding edge segment
+  int iu = (int)u;
+  if (SafeT == SafetyLevel::kSafe) {
+    iu = (iu < 0) ? 0 : (iu > mUmax ? mUmax : iu);
+  }
+  return getUtoKnotMap()[iu];
+}
+
+template <typename DataT>
+GPUdi() void Spline1DContainer<DataT>::setXrange(DataT xMin, DataT xMax)
```

Comment:
CUDA Compilation errors: (unfortunately the CI is currently not working, so I ran it manually)
```
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline1DSpec.h(256): warning: a __host__ function("o2::gpu::Spline1DContainer<DataT>::setXrange") redeclared with __device__, hence treated as a __host__ __device__ function

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline1D.h(162): error: argument list for class template "o2::gpu::Spline1DSpec" is missing

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2D.h(98): error: argument list for class template "o2::gpu::Spline2DSpec" is missing

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline1D.h(166): warning: "o2::gpu::Spline1D<DataT, YdimT>::operator o2::gpu::Spline1D<float, -999999> &() [with DataT=float, YdimT=-999999]" will not be called for implicit or explicit conversions
          detected during:
            instantiation of class "o2::gpu::Spline1D<DataT, YdimT> [with DataT=float, YdimT=-999999]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(70): here
            instantiation of class "o2::gpu::Spline2DContainer<DataT> [with DataT=float]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(226): here
            instantiation of class "o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent> [with DataT=float, YdimT=3, YisPositiveT=true, YisOneT=false, YisAbsent=false]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(318): here
            instantiation of class "o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false> [with DataT=float, YdimT=3]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2D.h(77): here
            instantiation of class "o2::gpu::Spline2D<DataT, YdimT> [with DataT=float, YdimT=3]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(250): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline1D.h(169): warning: "o2::gpu::Spline1D<DataT, YdimT>::operator const o2::gpu::Spline1D<float, -999999> &() const [with DataT=float, YdimT=-999999]" will not be called for implicit or explicit conversions
          detected during:
            instantiation of class "o2::gpu::Spline1D<DataT, YdimT> [with DataT=float, YdimT=-999999]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(70): here
            instantiation of class "o2::gpu::Spline2DContainer<DataT> [with DataT=float]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(226): here
            instantiation of class "o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent> [with DataT=float, YdimT=3, YisPositiveT=true, YisOneT=false, YisAbsent=false]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(318): here
            instantiation of class "o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false> [with DataT=float, YdimT=3]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2D.h(77): here
            instantiation of class "o2::gpu::Spline2D<DataT, YdimT> [with DataT=float, YdimT=3]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(250): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(268): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(268): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(276): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(269): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(269): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(276): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(270): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(270): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(276): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(271): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(271): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(276): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(287): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(287): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(276): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(292): error: no instance of overloaded function "o2::gpu::Spline1DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsentT>::interpolateU [with DataT=float, YdimT=12, YisPositiveT=true, YisOneT=false, YisAbsentT=false]" matches the argument list
            argument types are: (int, const o2::gpu::Spline1DContainer<float>::Knot, <error-type>, <error-type>, <error-type>, <error-type>, float, <error-type>)
            object type is: const TGridX1
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=3, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(276): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(268): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(268): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(240): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolate(DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false]" 
(406): here
            instantiation of "DataT o2::gpu::Spline2DSpec<DataT, 1, false, true, true, false>::interpolate(DataT, DataT) const [with DataT=float]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/dEdx/TPCdEdxCalibrationSplines.h(142): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(269): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(269): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(240): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolate(DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false]" 
(406): here
            instantiation of "DataT o2::gpu::Spline2DSpec<DataT, 1, false, true, true, false>::interpolate(DataT, DataT) const [with DataT=float]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/dEdx/TPCdEdxCalibrationSplines.h(142): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(270): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(270): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(240): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolate(DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false]" 
(406): here
            instantiation of "DataT o2::gpu::Spline2DSpec<DataT, 1, false, true, true, false>::interpolate(DataT, DataT) const [with DataT=float]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/dEdx/TPCdEdxCalibrationSplines.h(142): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(271): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(271): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(240): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolate(DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false]" 
(406): here
            instantiation of "DataT o2::gpu::Spline2DSpec<DataT, 1, false, true, true, false>::interpolate(DataT, DataT) const [with DataT=float]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/dEdx/TPCdEdxCalibrationSplines.h(142): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(287): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(287): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(240): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolate(DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false]" 
(406): here
            instantiation of "DataT o2::gpu::Spline2DSpec<DataT, 1, false, true, true, false>::interpolate(DataT, DataT) const [with DataT=float]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/dEdx/TPCdEdxCalibrationSplines.h(142): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(292): error: no instance of overloaded function "o2::gpu::Spline1DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsentT>::interpolateU [with DataT=float, YdimT=4, YisPositiveT=true, YisOneT=false, YisAbsentT=false]" matches the argument list
            argument types are: (int, const o2::gpu::Spline1DContainer<float>::Knot, <error-type>, <error-type>, <error-type>, <error-type>, float, <error-type>)
            object type is: const TGridX1
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(240): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolate(DataT, DataT, DataT *) const [with DataT=float, YdimT=1, YisPositiveT=true, YisOneT=false, YisAbsent=false]" 
(406): here
            instantiation of "DataT o2::gpu::Spline2DSpec<DataT, 1, false, true, true, false>::interpolate(DataT, DataT) const [with DataT=float]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/dEdx/TPCdEdxCalibrationSplines.h(142): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(268): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(268): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(307): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(269): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(269): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(307): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(270): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(270): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(307): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(271): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(271): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(307): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(287): error: expression must have a constant value
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(287): note: the value of variable "maxYdim4"
(249): here cannot be used as a constant
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(307): here

/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/Spline2DSpec.h(292): error: no instance of overloaded function "o2::gpu::Spline1DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsentT>::interpolateU [with DataT=float, YdimT=8, YisPositiveT=true, YisOneT=false, YisAbsentT=false]" matches the argument list
            argument types are: (int, const o2::gpu::Spline1DContainer<float>::Knot, <error-type>, <error-type>, <error-type>, <error-type>, float, <error-type>)
            object type is: const TGridX1
          detected during:
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, true, YisPositiveT, YisOneT, YisAbsent>::interpolateU(int, const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, YisPositiveT=true, YisOneT=false, YisAbsent=false, SafeT=o2::gpu::Spline1DContainer<float>::kSafe]" 
(377): here
            instantiation of "void o2::gpu::Spline2DSpec<DataT, YdimT, false, true, false, false>::interpolateU(const DataT *, DataT, DataT, DataT *) const [with DataT=float, YdimT=2, SafeT=(o2::gpu::Spline1DContainer<float>::SafetyLevel)1]" 
/home/qon/alice/O2/GPU/GPUTracking/Standalone/TPCFastTransformation/TPCFastSpaceChargeCorrection.h(307): here

20 errors detected in the compilation of "/home/qon/alice/O2/GPU/GPUTracking/Base/cuda/GPUReconstructionCUDA.cu".
```